### PR TITLE
User & Review model

### PR DESCRIPTION
**Pull request by rfmcmillan**
- added user model with tests
- added review model with tests

- let me know if there's anything you think we should change about them. If you guys don't object, I am going to spend this afternoon looking into JWT authentication for the user's password. If there's anything more pressing that you think I should get to, please let me know and I'll get to work on that.

### CODE REVIEW

**Review by YiruDing — APPROVED** [review]
It looks good to me!

**Review by kfless12 — APPROVED** [review]
Looks good,

for security purposes we may want to add a hook adjust to "beforeupdate" to check if the person is updating there admin status and if it is allowed but that can be addressed at a later date along with encrypting the password and saving only the encrypted version etc.

**Comment by rfmcmillan** [comment]
sounds good - when it seems like the right time I can go in and make those changes.

File: server/db/relationships.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ const Order = require('./models/Order')
 const Product = require('./models/Product')
 const ProductOrders = require('./models/ProductOrders')
 const Category = require('./models/Category')
+const User = require('./models/User')
+const Review = require('./models/Review')
 
 //once Product is imported
 //Category.belongsToMany(Product,  { through: 'productcategories' });
 //Product.belongsToMany(Category,  { through: 'productcategories' })
```
Comment (alejvillanueva) [inline server/db/relationships.js:12]: Could you please add the relationship for your models here
```suggestion
User.hasMany(Review)
Product.hasMany(Review)

Review.belongsTo(Product)
Review.belpongsTo(User)
```

**Review by alejvillanueva — CHANGES_REQUESTED** [review]
Everything looks could! Before merging, could you just add the relationship of your models in the relationship.js file?

File: server/db/relationships.js [hunk 2]
```diff
@@ -3,11 +3,21 @@ const Order = require('./models/Order')
...
+User.hasMany(Review)
```
Comment (alejvillanueva) [inline server/db/relationships.js:12]: Lines 12 - 20 have duplicate code which could cause an error

**Review by alejvillanueva — APPROVED** [review]
Great job!!

**Pull request merged**